### PR TITLE
Fixes #17308: Use cargo-deny to replace cargo-audit

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/17308

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: relay/sources/relayd/Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ description = "Rudder relay component"
 readme = "README.md"
 homepage = "https://www.rudder.io"
 repository = "https://github.com/Normation/rudder"
-license = "GPL3"
+license = "GPL-3.0-or-later"
```
Comment (amousset) [inline relay/sources/relayd/Cargo.toml:10]: Cargo actually expects an SPDX-formatted license name, things are converging somehow.

Comment (peckpeck) [reply relay/sources/relayd/Cargo.toml:10]: but or-later is not the sale a gpl-3

Comment (amousset) [reply relay/sources/relayd/Cargo.toml:10]: Indeed, and previous value was actually wrong (https://github.com/Normation/rudder/blob/master/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/api/DataStructures.scala#L10 or any GPL header in Rudder).

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**